openassessment/assessment/models/staff.py

```diff
@@ -51,6 +51,35 @@ def is_cancelled(self):
         return bool(self.cancelled_at)
 
     @classmethod
+    def get_workflow_statistics(cls, course_id, item_id):
+        """
+        Returns the number of graded, ungraded, and in-progress submissions there are for staff grading.
+
+        Args:
+            course_id (str): The course that this problem belongs to
+            item_id (str): The student_item (problem) that we want to know statistics about.
+
+        Returns:
+            dict: a dictionary that contains the following keys: 'graded', 'ungraded', and 'in-progress'
+        """
+        timeout = (now() - cls.TIME_LIMIT).strftime("%Y-%m-%d %H:%M:%S")
+        ungraded = cls.objects.filter(
+            models.Q(grading_started_at=None) | models.Q(grading_started_at__lte=timeout),
+            course_id=course_id, item_id=item_id, grading_completed_at=None, cancelled_at=None
+        ).count()
+
+        in_progress = cls.objects.filter(
+            course_id=course_id, item_id=item_id, grading_completed_at=None, cancelled_at=None,
+            grading_started_at__gt=timeout
+        ).count()
+
+        graded = cls.objects.filter(
+            course_id=course_id, item_id=item_id, cancelled_at=None
+        ).exclude(grading_completed_at=None).count()
```

Comment:
Instead of making 3 separate queries, could we get 2 of the 3 and calculate the remainder by subtracting from `cls.objects.all().count()`? Might make for a small performance boost, but I'm not sure that's a safe assumption to make.